packages/catalog-process/src/services/catalogService.ts

```diff
@@ -1334,6 +1335,71 @@ export function catalogServiceBuilder(
       return updatedEService;
     },
 
+    async updateDraftDescriptorInstance(
+      eserviceId: EServiceId,
+      descriptorId: DescriptorId,
+      seed: catalogApi.UpdateEServiceDescriptorInstanceSeed,
+      { authData, correlationId, logger }: WithLogger<AppContext>
+    ): Promise<EService> {
+      logger.info(
+        `Updating draft Descriptor ${descriptorId} for EService ${eserviceId} Instance`
+      );
+
+      const eservice = await retrieveEService(eserviceId, readModelService);
+      await assertRequesterIsDelegateProducerOrProducer(
+        eservice.data.producerId,
+        eservice.data.id,
+        authData,
+        readModelService
+      );
+
+      if (eservice.data.templateRef === undefined) {
+        throw eServiceNotAnInstance(eserviceId);
+      }
+
+      const descriptor = retrieveDescriptor(descriptorId, eservice);
+
+      //  Descriptor in state WaitingForApproval can be updated
+      if (descriptor.state !== descriptorState.draft) {
```

Comment:
The comment says that descriptors in WaitingForApproval can be updated, but the check allows only Draft descriptors. Maybe the comment is not needed?
(I see a similar comment in many methods of this file, with the same contradiction - could you remove it everywhere?)